### ISSUE TITLE
Device version check is wrong?

### ISSUE THREAD

**Issue by ignus2** [Oblomov/clinfo#94]
Certain device features are not printed if the device doesn't meet a given OpenCL version requirement. For example the CL_DEVICE_OPENCL_C_VERSION is not printed for devices that only report themselves as OpenCL 1.0 compatible.

However I believe this check is wrong. According to the OpenCL spec: https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_DEVICE_OPENCL_C_VERSION  

> CL_DEVICE_OPENCL_C_VERSION
> missing before version 1.1 and deprecated by version 3.0.

Which might imply we need to check the "device" version before querying this value, however the spec says:

> Missing before major.minor: Features that were introduced in version major.minor. Implementations of an earlier version of OpenCL will not provide these features.

It specifically says **_Implementation_**. And looking at CL_PLATFORM_VERSION vs CL_DEVICE_VERSION:

> CL_PLATFORM_VERSION: OpenCL version string. Returns the OpenCL version supported by the implementation.

I have a test card 8800 GT where platform version is 1.1 but device version is 1.0, so the CL_DEVICE_OPENCL_C_VERSION query is skipped, however it can be queried and it prints the correct info.

So in summary, I think wherever the spec says "missing before" should be interpreted as missing before a given platform version, not device version.

**Comment by ignus2** [comment]
I was afraid the reason was that, I also wanted to verify if it's a known thing, and apparently it is.

Thanks, the `-a` flag works perfectly.

**Comment by Oblomov** [comment]
Glad `-a` works for you. FWIW, I also wish I didn't have to work around platform bugs, but such is life 8-)
